internal/databases/mysql/stream_fetch_handler.go

```diff
@@ -45,21 +45,66 @@ func FetchLogs(folder storage.Folder, backup *internal.Backup) error {
 			backupUploadTime = binlog.GetLastModified()
 		}
 	}
-	dstFolder, fetchedBinlogs, err := internal.FetchLogs(folder, backupUploadTime, BinlogFetchSettings{})
+
+	logsToFetch, err := internal.GetLogsCoveringInterval(folder, backupUploadTime, nil)
 	if err != nil {
 		return err
 	}
-	return createIndexFile(dstFolder, fetchedBinlogs)
+	endTS, logDstFolder, err := internal.GetOperationLogsSettings(BinlogFetchSettings{}.GetEndTsEnv(), BinlogFetchSettings{}.GetDstEnv())
+
+	if err != nil {
+		return err
+	}
+
+	fetchedBinlogs, err := filterBinlogByHeaderTimestamp(folder, logsToFetch, logDstFolder, endTS)
+
+	if err != nil {
+		return err
+	}
+
+	return createIndexFile(logDstFolder, fetchedBinlogs)
+}
+
+func filterBinlogByHeaderTimestamp(folder storage.Folder, logsToFetch []storage.Object, logDstFolder string, endTS *time.Time) ([]string, error) {
```

Comment:
And if you will pass it as a filter func you should to return []storage.Object instead of []string